be/src/exec/hdfs_scanner.cpp

```diff
@@ -81,7 +81,14 @@ bool HdfsScannerParams::is_lazy_materialization_slot(SlotId slot_id) const {
 Status HdfsScanner::init(RuntimeState* runtime_state, const HdfsScannerParams& scanner_params) {
     _runtime_state = runtime_state;
     _scanner_params = scanner_params;
+
+    RETURN_IF_ERROR(_build_hash_joiner_for_mor_if_needed());
     Status status = do_init(runtime_state, scanner_params);
+
+    if (!scanner_params.equality_slots.empty()) {
```

Comment:
can you do `build_ht` and `enter_probe_phase` in `build_hash_joiner_for_mor_if_needed`? 

Plus, I think we'd better to think MOR as default mode, so rename it to 

`build_hash_joiner_for_mor`.